core/arch/arm/kernel/ree_fs_ta.c

```diff
@@ -191,6 +261,10 @@ static TEE_Result ree_fs_ta_open(const TEE_UUID *uuid,
 			goto error_free_hash;
 		}
 
+		res = check_update_version(&bs_hdr);
```

Comment:
So we might save a version although the TA is not yet fully validated?

Reply:
Yeah and that might lead to a kind of DoS issue for TA which is expected to work. So yes, will move this check after call to `check_digest()`.